docs/cluster-adoption.md

```diff
@@ -0,0 +1,49 @@
+# Cluster Adoption
```

Comment:
We should link to this doc from... somewhere. Do we consider adoption a dev-only thing? If so, perhaps the suggested link from developing.md is sufficient. Otherwise maybe from using-hive.md?

Reply:
> Do we consider adoption a dev-only thing?

Looking at the [card](https://issues.redhat.com/browse/HIVE-1566) I can see the answer is "not anymore", so...

Reply:
I'm thinking top level readme. Thought looking at what's there, maybe using hive is a better fit.